### PR TITLE
core: allow empty MPT Leaf values

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Port https://github.com/neo-project/neo-modules/pull/634, waiting for the reference PR to settle.

### CODE REVIEW

File: pkg/core/mpt/compat_test.go [hunk 1]
```diff
@@ -72,10 +72,11 @@ func TestCompatibility(t *testing.T) {
 
 		require.Equal(t, mainTrie.root.Hash(), tr.root.Hash())
 		require.Error(t, tr.Put(nil, []byte{0x01}))
-		require.NoError(t, tr.Put([]byte{0x01}, nil))
```
Comment (fyrchik) [inline pkg/core/mpt/compat_test.go:75]: These tests are also altered in C# PR, we need to follow. For example this line should be changed to
`require.NoError(t, tr.Put([]byte{0x01}, []byte{}))`
https://github.com/neo-project/neo-modules/pull/634/files#diff-a7cab0a54353f7147388e535326be4adc8bd3ba2c26e0b48d54ab049566359cfR197

**Review by fyrchik — COMMENTED** [review]
It would also be nice to have more tests for empty value in `mpt/trie_test.go` (for all types of nodes). Tests in `mpt/batch_test.go` just test that the batch implementation behaves the same way as the simple one, so they both can be incorrect.

**Comment by fyrchik** [comment]
Other than tests, LGTM.

File: pkg/core/mpt/compat_test.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/core/mpt/compat_test.go:75]: It is already replaced, but a few lines below (after the check that root hash of the etalon tree is equal to the actual one). In C# they also do not change the etalon tree and perform addition of the empty value after this check.

Comment (AnnaShaleva) [reply pkg/core/mpt/compat_test.go:75]: As a result, this line checks that empty value can be safely added to the tree, but do not perform root hashes comparison. C# code does the same.

**Review by roman-khimov — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]
